domain.c

```diff
@@ -1238,6 +1347,13 @@ order_by_key(const void *a, const void *b)
     return 0;
```

Comment:
I'll add back the local sorting routine but disable it with a flag. I worry one may prefer local sorting in some cases (I can't yet think of any).